command/ca/provisioner/update.go

```diff
@@ -0,0 +1,875 @@
+		if err != nil {
+			return err
+		}
+		jwk, jwe, err = jose.GenerateDefaultKeyPair(pass)
+		if err != nil {
+			return err
+		}
+	} else {
+		if ctx.IsSet("private-key") && ctx.IsSet("no-private-key") {
+			return errs.IncompatibleFlagWithFlag(ctx, "private-key", "no-private-key")
+		}
+
+		if ctx.IsSet("public-key") {
+			jwkFile := ctx.String("public-key")
+			jwk, err = jose.ParseKey(jwkFile)
+			if err != nil {
+				return errs.FileError(err, jwkFile)
+			}
+
+			// Only use asymmetric cryptography
+			if _, ok := jwk.Key.([]byte); ok {
+				return errors.New("invalid JWK: a symmetric key cannot be used as a provisioner")
+			}
+			// Create kid if not present
+			if jwk.KeyID == "" {
+				jwk.KeyID, err = jose.Thumbprint(jwk)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		if ctx.IsSet("private-key") {
+			jwkFile := ctx.String("private-key")
+			b, err := os.ReadFile(jwkFile)
+			if err != nil {
+				return errors.Wrapf(err, "error reading %s", jwkFile)
+			}
+
+			// Attempt to parse private key as Encrypted JSON.
+			// If this operation fails then either,
+			//   1. the key is not encrypted
+			//   2. the key has an invalid format
+			//
+			// Attempt to parse as decrypted private key.
+			jwe, err = jose.ParseEncrypted(string(b))
+			if err != nil {
+				privjwk, err := jose.ParseKey(jwkFile)
+				if err != nil {
+					return errs.FileError(err, jwkFile)
+				}
+
+				if privjwk.IsPublic() {
+					return errors.New("invalid jwk: private-key is a public key")
+				}
+
+				// Encrypt JWK
+				opts := []jose.Option{}
+				if ctx.IsSet("password-file") {
+					opts = append(opts, jose.WithPasswordFile(ctx.String("password-file")))
+				}
+				jwe, err = jose.EncryptJWK(privjwk, opts...)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	if jwk != nil {
+		jwkPubBytes, err := jwk.MarshalJSON()
+		if err != nil {
+			return errors.Wrap(err, "error marshaling JWK")
+		}
+		details.PublicKey = jwkPubBytes
+	}
+
+	if ctx.Bool("no-private-key") {
+		details.EncryptedPrivateKey = nil
+	} else if jwe != nil {
+		jwePrivStr, err := jwe.CompactSerialize()
+		if err != nil {
+			return errors.Wrap(err, "error serializing JWE")
+		}
+		details.EncryptedPrivateKey = []byte(jwePrivStr)
+	}
+
+	return nil
+}
+
+func updateACMEDetails(ctx *cli.Context, p *linkedca.Provisioner) error {
+	data, ok := p.Details.GetData().(*linkedca.ProvisionerDetails_ACME)
+	if !ok {
+		return errors.New("error casting details to ACME type")
+	}
+	details := data.ACME
+	if ctx.IsSet("force-cn") {
+		details.ForceCn = ctx.Bool("force-cn")
+	}
+	requireEABSet := ctx.IsSet("require-eab")
+	disableEABSet := ctx.IsSet("disable-eab")
+	if requireEABSet && disableEABSet {
+		return errs.IncompatibleFlagWithFlag(ctx, "require-eab", "disable-eab")
+	}
+	if requireEABSet {
+		details.RequireEab = ctx.Bool("require-eab")
+	}
+	if disableEABSet {
+		details.RequireEab = false
+	}
```

Comment:
Not saying that we should do it, but if we want to reduce boolean flags, we can do  `--require-eab=false` and remove `--disable-eab`. 

In your implementation `--disable-eab` == `--disable-eab=false`, it should be ok.